app/assets/javascripts/views/schedule.js

```diff
@@ -53,6 +54,24 @@ Yacs.views.schedule = function (data) {
     crnListElement.textContent = 'CRNs: ' + scheduleData.crns.join(', ');
   }
 
+  /* this is before `if(data.schedules.length==0)` because clear selection should */
+  /* still work even if courses conflict and there are zero possible schedules    */
+  Yacs.on('click', clearButtonElement, function () {
+    /* clear if the user has any selections */  
+    if(Yacs.user.getSelections().length != 0){
+      Yacs.user.clearSelections();
+      /* copied from yacs.js schedule button handler */
+      Yacs.models.schedules.query({ section_ids: Yacs.user.getSelectionsRaw(),
```

Comment:
If there are no sections selected, then there is no need for a query. simply pass the schedule view empty data. additionally, try to avoid copying code in general.